### ISSUE TITLE
the currentFreq won't exceed base freq 

### ISSUE THREAD

**Issue by Reqeique** [oshi/oshi#2187]
was  using this library for while but found that even at max frequency (3.2 ghz) the `currentFreq` (from `CentralProccesor` class) reading was frozen to the base frequency (1.58 ghz) but if the speed was below the `baseFreq` it works completely fine, 
any way to resolve this issue ?
### Machine
1. Windows 64 bit
2. I5- 8250u

**Comment by dbwiddis** [comment]
> any way to resolve this issue ?

Probably not, but if you can think of one, I'm open to it!

Accurately measuring CPU frequency in OSHI is difficult and there have been quite a few previous issues discussing this.  I'll try to sum up the points:
 - There is limited utility to measuring instantaneous CPU frequency. By the time you've measured it and reported it, it has likely changed.
   - Your power settings have an impact.  My Windows machine rarely throttles down its processors when plugged in, making measurement difficult.  I had to unplug to see any changes in testing, and they bounced up and down depending on which processor was doing work.
 - The Windows API provides very little information on CPU frequency.  The official Win32API supports a call to [`CallNtPowerInformation`](CallNtPowerInformation) with the `ProcessorInformation` information level. This returns a [`PROCESSOR_POWER_INFORMATION`](https://docs.microsoft.com/en-us/windows/win32/power/processor-power-information-str) which returns a "CurrentMhz" value which the documents indicate is "the maximum specified processor clock frequency multiplied by the current processor throttle.  
   - This seems to report only a few fixed numbers, the max frequency and some lower throttled frequency (e.g., either 2.3 GHz or 1.4 GHz on some processors, sometimes, if you time it right.)
 - Windows Performance Counters have more information. "Processor Performance" counters have a "ProcessorFrequency" value which seems to exactly match the above (NTPowerInformation) API, but they also have a "PercentofMaximumFrequency" value which appears to be more fine-grained, and probably an average rather than the instantaneous and somewhat "binary" high/low measure.  _This is what OSHI currently reports._
 - The PerfMon counters do have a "PercentProcessorPerformance" value which reports "work completed" and gives some higher level of detail of "boost" performance.   The equivalent to this value is used in the "CPU Utility" calculations for CPU load (because that's what it measures).
   - It may be reasonable to use the "percent frequency" value when at <100% load and "CPU Utility" value when > 100% load, but that's an end user implementation decision; the library is reporting all it can.

If you don't limit yourself to Java, your options improve.  Intel processors have a low level instruction to directly query the processor's frequency.  But you can't run the necessary assembly language instructions from Java.  And you need to have a different instruction for different processors.  Some apps like CpuZ do this... and if you really want to measure your CPU you might want to use one of them, not any Java-based application.

More info on #996.

In summary:
 - I believe OSHI is reporting the best possible source for CPU frequency when it's throttled
 - I believe OSHI's (optional) CPU Utility calculation provides a reasonable estimate of boost performance at higher loads
 - I have tried just about everything I can try and don't know of any other way to get this information in Java and am welcome to any additional input if you are aware of a data source I don't know about.

**Comment by Reqeique** [comment]
Thanks for the reply  @dbwiddis 
I don't really have much knowledge about low-level language and it's api can't really yield any helpful information for you 
but the reply should be enough to explain why it's super hard for me to find any helpful information on gathering frequency of the CPU using Jvm based langs in the internets 
fortunately in the case of mine the project run's on the JVM thus can't use  assembly-lang even if I do so the implementation wouldn't worth the time
I had also planned to use JNI but after [this](https://stackoverflow.com/questions/8351944/finding-out-the-cpu-clock-frequency-per-core-per-processor) wasn't worth the effort
In my opinion for the time being I should just use oshi
Thanks
